core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -472,15 +471,15 @@ class ZkUtils(zkClientWrap: ZooKeeperClientWrapper,
   }
 
   def leaderAndIsrZkData(leaderAndIsr: LeaderAndIsr, controllerEpoch: Int): String = {
-    Json.encode(Map("version" -> 1, "leader" -> leaderAndIsr.leader, "leader_epoch" -> leaderAndIsr.leaderEpoch,
-                    "controller_epoch" -> controllerEpoch, "isr" -> leaderAndIsr.isr))
+    Json.encodeToJsonString(Map("version" -> 1, "leader" -> leaderAndIsr.leader, "leader_epoch" -> leaderAndIsr.leaderEpoch,
+                    "controller_epoch" -> controllerEpoch, "isr" -> leaderAndIsr.isr.asJava))
   }
 
   /**
    * Get JSON partition to replica map from zookeeper.
    */
-  def replicaAssignmentZkData(map: Map[String, Seq[Int]]): String = {
-    Json.encode(Map("version" -> 1, "partitions" -> map))
+  def replicaAssignmentZkData(partitionMap: Map[Int, Seq[Int]]): String = {
+    Json.encodeToJsonString(Map("version" -> 1, "partitions" -> (partitionMap.map(e => e._1.toString -> e._2.asJava).asJava)))
```

Comment:
Similar to `AdminUtils`, I would leave `ZkUtils` unchanged for now.

Reply:
Noted.